### PR TITLE
Blog post: Tuist now supports detecting flaky tests

### PR DESCRIPTION
**Pull request by pepicrft**

### CODE REVIEW

File: src/content/posts/2024/07/10/detecting-flaky-tests.mdx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Tuist now supports detecting flaky tests"
+categories: ["Swift", "XcodeProj", "Xcode"]
```
Comment (fortmarek) [inline src/content/posts/2024/07/10/detecting-flaky-tests.mdx]:
```suggestion
categories: ["Tests", "Swift", "XcodeProj", "Xcode"]
```

Would be good to always add a category to tie it with the feature the blog post is about (if any). Adding the `Tuist Projects` category doesn't make much sense as most need that, so I'd only use the categories for `Test`, `Cache`, and soon `Share`

File: src/content/posts/2024/07/10/detecting-flaky-tests.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+excerpt: "Tuist now supports detecting flaky tests. Learn how it works and how it can help you ship more reliable software"
+author: pepicrft
+---
+
+When developing software, it's standard practice to write tests to ensure that code behaves as expected. In the Apple ecosystem, developers have been using [XCTest](https://developer.apple.com/documentation/xctest) to write tests for their Swift and Objective-C codebases. Recently, [Swift Testing](https://developer.apple.com/xcode/swift-testing/) was introduced, addressing many of XCTest's shortcomings and advancing the state of the art in testing Swift code.
+
+## Understanding Flaky Tests
+
+If you've written tests before, you might know that sometimes tests fail even though the code is correct. These are often referred to as ["flaky tests"](https://docs.gitlab.com/ee/development/testing_guide/flaky_tests.html). In other words, flaky tests are not deterministic. They might fail due to race conditions or dependencies on the environment in which they run.
```
Comment (fortmarek) [inline src/content/posts/2024/07/10/detecting-flaky-tests.mdx]: nit: Flaky tests can very much signal that the code is `incorrect`, so maybe worth rephrasing the first sentence.

Comment (fortmarek) [inline src/content/posts/2024/07/10/detecting-flaky-tests.mdx]:
```suggestion
If you've written tests before, you might know that sometimes tests fail even though the code is correct. These are often referred to as ["flaky tests"](https://docs.gitlab.com/ee/development/testing_guide/unhealthy_tests.html#flaky-tests). In other words, flaky tests are not deterministic. They might fail due to race conditions or dependencies on the environment in which they run.
```

Comment (fortmarek) [reply src/content/posts/2024/07/10/detecting-flaky-tests.mdx]: The current URL leads to an empty page

**Review by fortmarek — APPROVED** [review]
Well-written! 🙌

File: src/content/posts/2024/07/10/detecting-flaky-tests.mdx (hunk 1 above)
Comment (pepicrft) [reply src/content/posts/2024/07/10/detecting-flaky-tests.mdx]: Good idea! I just noticed I've been using categories as tags, but they should be different things. Let's revisit this when we migrate to the new website. For now I'll add tests there.

**Pull request merged**